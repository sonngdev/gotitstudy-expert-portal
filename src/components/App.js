import React from 'react';
import { TopBar } from './TopBar';
import { ChatArea } from './Body/ChatArea';
import { ExplanationArea } from './Body/ExplanationArea';

export function App() {
  return (
    <>
      <TopBar />

      <div className="container-app Container u-paddingTopSmall">
        <div className="Grid Grid--smallGutter">
          <div className="u-size1of2">
            <ChatArea />
          </div>

          <div className="u-size1of2">
            <ExplanationArea />
          </div>
        </div>
      </div>
    </>
  );
}
