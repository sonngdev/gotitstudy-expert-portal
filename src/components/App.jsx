import React, { Fragment } from 'react';
import { TopBar } from './TopBar';
import { ChatArea } from './Body/ChatArea';
import { ExplanationArea } from './Body/ExplanationArea';

export function App() {
  return (
    <Fragment>
      <TopBar />

      <div className="container-app">
        <ChatArea />

        <ExplanationArea />
      </div>
    </Fragment>
  );
}
