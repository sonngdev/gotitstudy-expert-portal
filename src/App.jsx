import React from 'react';
import { TopBar } from './components/TopBar';

function App() {
  return (
    <TopBar />
  );
}

export default App;
