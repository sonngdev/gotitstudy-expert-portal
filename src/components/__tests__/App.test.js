import React from 'react';
import { shallow } from 'enzyme';
import { App } from '../App';

describe('<App />', () => {
  let wrapper;
  let container;

  beforeEach(() => {
    wrapper = shallow(<App />);
    container = wrapper.find('.container-app');
  });

  it('renders', () => {
    const r = () => shallow(<App />);
    expect(r).not.toThrow();
  });

  it('contains a TopBar', () => {
    const topbar = wrapper.find('TopBar');
    expect(topbar.exists()).toBe(true);
  });

  it('contains a container for main content', () => {
    expect(container.exists()).toBe(true);
  });

  it('has a ChatArea in its container', () => {
    const chat = container.find('ChatArea');
    expect(chat.exists()).toBe(true);
  });

  it('has a ExplanationArea in its container', () => {
    const explanation = container.find('ExplanationArea');
    expect(explanation.exists()).toBe(true);
  });
});
