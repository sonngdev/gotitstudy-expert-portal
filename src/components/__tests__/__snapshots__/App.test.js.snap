// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders 1`] = `
"<Fragment>
  <TopBar />
  <div className=\\"container-app Container u-paddingTopSmall\\">
    <div className=\\"Grid Grid--smallGutter\\">
      <div className=\\"u-size1of2\\">
        <ChatArea />
      </div>
      <div className=\\"u-size1of2\\">
        <ExplanationArea />
      </div>
    </div>
  </div>
</Fragment>"
`;
